ledger/src/blockstore.rs

```diff
@@ -7166,51 +7154,46 @@ pub mod tests {
             .unwrap();
 
         let slot_meta = blockstore.meta(0).unwrap().unwrap();
-        let shred5 = shreds[5].clone();
-
-        // Ensure that an empty shred (one with no data) would get inserted. Such shreds
-        // may be used as signals (broadcast does so to indicate a slot was interrupted)
-        // Reuse shred5's header values to avoid a false negative result
-        let empty_shred = Shred::new_from_data(
-            shred5.slot(),
-            shred5.index(),
-            {
-                let parent_offset = shred5.slot() - shred5.parent().unwrap();
-                parent_offset as u16
-            },
-            &[], // data
-            ShredFlags::LAST_SHRED_IN_SLOT,
-            0, // reference_tick
-            shred5.version(),
-            shred5.fec_set_index(),
-        );
+
+        // make a "blank" FEC set that would normally be used to terminate the block
+        let terminator = shredder
+            .entries_to_shreds(
+                &keypair,
+                &[],
+                true,
+                None, //chained_merkle_root
+                6,    // next_shred_index,
+                6,    // next_code_index
+                true,
+                &rsc,
+                &mut ProcessShredsStats::default(),
+            )
+            .0;
+
         assert!(blockstore.should_insert_data_shred(
-            &empty_shred,
+            &terminator[0],
             &slot_meta,
             &HashMap::new(),
             max_root,
             None,
             ShredSource::Repaired,
             &mut Vec::new(),
         ));
+        let term_last_idx = terminator.last().unwrap().index() as usize;
```

Comment:
Isn't this skipping to some index much farther in the future (like 38?). I mention this because the comment below indicates we are just skipping over shred 7

Reply:
Yeah shred 7 is gone now. Sadly AI does not keep comments in sync yet. rm comment.

Reply:
the one with flag should be last data shred in fec set, so yes, terminator.last() should be the one with flag set.